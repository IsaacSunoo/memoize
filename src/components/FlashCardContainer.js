import React, { Component }from 'react';
import FlashCard from './FlashCard';
import Header from './layout/Header';

class FlashCardContainer extends Component {
    constructor() {
        super();
        this.state = {
            rangeValue: 1,
        }
    }

    determineRangeValue = (e) => {
        this.setState({
            rangeValue: e.target.value,
        })
    }

    render() {
        return (
            <div>
                <Header />
                <h2 className='flash-card-title'>FlashCards</h2>
                <section className='flash-card-section'>
                    <div className='flash-card-container'>
                        {
                            this.props.prototype.map(proto => {
                                if (proto.id === parseInt(this.state.rangeValue)) {
                                    return <FlashCard key={proto.id} proto={proto} />
                                } else {
                                    return '';
                                }
                            })
                        }
                    </div>
                        <input type='range' min='1' max='30' className='range-bar' value={this.state.rangeValue} onChange={this.determineRangeValue}></input>
                        <p className='card-number'>{this.state.rangeValue} / 30</p>
                </section>
            </div>
        )
    }
}

export default FlashCardContainer;
