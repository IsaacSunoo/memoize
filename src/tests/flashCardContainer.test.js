import React from 'react';
import FlashCardContainer from '../components/FlashCardContainer';
import { shallow } from 'enzyme';

const mockData = [{
    "id": 1,
    "prototype": "shift()",
    "answer": "Removes and returns the first item in an Array",
    "incorrectChoices": [
        "pop()",
        "push()",
        "unshift()"
    ],
    "method": "mutator",
    "link": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/shift"
},
{
    "id": 2,
    "prototype": "copyWithin()",
    "answer": "Shallow copies part of an array to another location in the same array and returns it, without modifying its size",
    "incorrectChoices": [
        "splice()",
        "findIndex()",
        "concat()"
    ],
    "method": "mutator",
    "link": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/copyWithin"
}
]
describe('FlashCardContainer', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(
            <FlashCardContainer
                prototype={mockData}
            />
        )
    })

    it('should match the snapshot with all data passed in correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should have the proper default state', () => {
        expect(wrapper.state()).toEqual({ rangeValue: 1 });
    });

    it('should determine the range value', () => {
        expect(wrapper.state('rangeValue')).toEqual(1);
        wrapper.find('.range-bar').simulate('change', { target: { value: 2 } });
    });



});