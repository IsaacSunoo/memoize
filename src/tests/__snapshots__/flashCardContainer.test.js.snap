// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlashCardContainer should match the snapshot with all data passed in correctly 1`] = `
<div>
  <Header />
  <h2
    className="flash-card-title"
  >
    FlashCards
  </h2>
  <section
    className="flash-card-section"
  >
    <div
      className="flash-card-container"
    >
      <FlashCard
        key="1"
        proto={
          Object {
            "answer": "Removes and returns the first item in an Array",
            "id": 1,
            "incorrectChoices": Array [
              "pop()",
              "push()",
              "unshift()",
            ],
            "link": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/shift",
            "method": "mutator",
            "prototype": "shift()",
          }
        }
      />
    </div>
    <input
      className="range-bar"
      max="30"
      min="1"
      onChange={[Function]}
      type="range"
      value={1}
    />
    <p
      className="card-number"
    >
      1
       / 30
    </p>
  </section>
</div>
`;
